### PR TITLE
Closes #487: Add execution info for module transactions

### LINKED ISSUE #487: `TransactionSummary::ExecutionInfo` add new variant for `ModuleTxs`
## Feature

`TransactionSummary::ExecutionInfo` only has data relevant to `MultisigTransactions`. We need to introduce polymorphism and add a variant for `ModuleTransactions` containing information on the module that triggered the transaction.  

## Proposal

Should have the same variants as https://github.com/gnosis/safe-client-gateway/blob/main/src/models/service/transactions/details.rs#L145

### PR DESCRIPTION
**Pull request by rmeissner**
Closes #487: Add execution info for module transactions

### CODE REVIEW

File: src/models/service/transactions/summary.rs [hunk 1]
```diff
@@ -138,16 +138,29 @@ pub struct TransactionSummary {
     pub safe_app_info: Option<SafeAppInfo>,
 }
 
+#[derive(Serialize, Debug, PartialEq)]
+#[serde(tag = "type", rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum ExecutionInfo {
+    Multisig(MultisigExecutionInfo),
+    Module(ModuleExecutionInfo),
+}
```
Comment (jpalvarezl) [inline src/models/service/transactions/summary.rs:146]: We would need to adjust the sample json in this file to contain ` "type" : "MULTISIG_EXECUTION_INFO"` If you are tight on time, we can open an issue to update the docs after we've finished all the api breaking changes.

**Review by jpalvarezl — APPROVED** [review]
Only comment would be that we need to update docs

File: src/models/converters/transactions/tests/summary.rs [hunk 2]
```diff
@@ -152,6 +166,61 @@ async fn module_tx_to_summary_transaction_failed() {
...
+    mock_info_provider.expect_safe_info().times(0);
+    mock_info_provider.expect_token_info().times(0);
+    mock_info_provider
+        .expect_address_ex_from_contracts()
+        .times(1)
+        .return_once(move |address| {
+            Ok(AddressEx {
+                value: address.to_string(),
+                name: Some(format!("{}_name", address)),
+                logo_url: None,
```
Comment (jpalvarezl) [inline src/models/converters/transactions/tests/summary.rs:183]: This should be adjusted after you rebase. Some how the PR is still mergable :/

**Review by jpalvarezl — CHANGES_REQUESTED** [review]
Just to prevent some dodgy stuff from happening after rebase

File: src/models/converters/transactions/tests/summary.rs (hunk 2 above)
Comment (fmrsabino) [reply src/models/converters/transactions/tests/summary.rs:183]: Yeah the checks were done on a state where the code that was added didn't have any conflicts (currently it still doesn't have conflicts so we should rebase).

**Review by fmrsabino — CHANGES_REQUESTED** [review]

**Review by jpalvarezl — APPROVED** [review]

**Pull request merged**